pkg/vault/reconcile.go

```diff
@@ -194,22 +186,3 @@ func DataInSecret(data map[string]interface{}, path string) bool {
 	}
 	return true
 }
-
-// Yaml unmarshal limitation causes nested options objects to be decode as strings with json format
-// ex: `{"foo": "bar"}`
-// UnmarshalJsonObj performs unmarshal of jsons strings
-func UnmarshalJsonObj(key string, obj interface{}) (map[string]interface{}, error) {
```

Comment:
This function was originally implemented with work to support oidc roles. Once use case for this functionality also arose within entity implementation, I elected to extract definition from role package and generalize within pkg/utils